### ISSUE TITLE
cargo install --locked doesn't honour the lockfile for a git source

### ISSUE THREAD

**Issue by Nemo157** [rust-lang/cargo#7495]
It appears that `cargo install --locked` doesn't honour the lockfile for a `git` source (with a virtual manifest, in case that's relevant).

```console
> cargo install -Z install-upgrade alacritty --git https://github.com/thunderseethe/alacritty --rev 853619f3dd8cb3a9de25beb1ac5e8ab6358d6986 --locked
    Updating git repository `https://github.com/thunderseethe/alacritty`
  Installing alacritty v0.3.3 (https://github.com/thunderseethe/alacritty?rev=853619f3dd8cb3a9de25beb1ac5e8ab6358d6986#853619f3)
    Updating crates.io index
error: failed to compile `alacritty v0.3.3 (https://github.com/thunderseethe/alacritty?rev=853619f3dd8cb3a9de25beb1ac5e8ab6358d6986#853619f3)`, intermediate artifacts can be found at `/tmp/cargo-installHN73vr`

Caused by:
  multiple packages link to native library `freetype`, but a native library can be linked only once

package `freetype-sys v0.7.1`
    ... which is depended on by `freetype-rs v0.19.1`
    ... which is depended on by `font v0.1.0 (https://github.com/thunderseethe/alacritty?rev=853619f3dd8cb3a9de25beb1ac5e8ab6358d6986#853619f3)`
    ... which is depended on by `alacritty_terminal v0.3.3 (https://github.com/thunderseethe/alacritty?rev=853619f3dd8cb3a9de25beb1ac5e8ab6358d6986#853619f3)`
    ... which is depended on by `alacritty v0.3.3 (https://github.com/thunderseethe/alacritty?rev=853619f3dd8cb3a9de25beb1ac5e8ab6358d6986#853619f3)`
links to native library `freetype`

package `servo-freetype-sys v4.0.3`
    ... which is depended on by `freetype v0.4.1`
    ... which is depended on by `harfbuzz-sys v0.3.2`
    ... which is depended on by `harfbuzz_rs v1.0.0`
    ... which is depended on by `font v0.1.0 (https://github.com/thunderseethe/alacritty?rev=853619f3dd8cb3a9de25beb1ac5e8ab6358d6986#853619f3)`
    ... which is depended on by `alacritty_terminal v0.3.3 (https://github.com/thunderseethe/alacritty?rev=853619f3dd8cb3a9de25beb1ac5e8ab6358d6986#853619f3)`
    ... which is depended on by `alacritty v0.3.3 (https://github.com/thunderseethe/alacritty?rev=853619f3dd8cb3a9de25beb1ac5e8ab6358d6986#853619f3)`
also links to native library `freetype`
```

If I then install from the checkout produced it succeeds fine

```console
> cargo install -Z install-upgrade alacritty --path ~/.cargo/git/checkouts/alacritty-abdebd0a1ed9f82f/853619f/alacritty --locked
  Installing alacritty v0.3.3 (/home/wim/.cargo/git/checkouts/alacritty-abdebd0a1ed9f82f/853619f/alacritty)
    Updating crates.io index
   Compiling autocfg v0.1.6
[...]
```

(somewhat related to #4263, but that appears to be from before `install --locked` was the explicit opt-in).

**Comment by chrisduerr** [comment]
This is a duplicate of https://github.com/rust-lang/cargo/issues/5495.

The problem is not with the lockfile, but that the virtual manifest is ignored.

**Comment by alexcrichton** [comment]
Closing as a duplicate of https://github.com/rust-lang/cargo/issues/5495
